dlib/smart_pointers.h

```diff
@@ -13,10 +13,21 @@
 #pragma GCC warning "smart_pointers.h is included which will fail to compile under C++17"
 #endif
 
-#include "smart_pointers/scoped_ptr.h"
+#include <memory>
+
 #include "smart_pointers/shared_ptr.h"
 #include "smart_pointers/weak_ptr.h"
 
+namespace dlib {
+  // Template alias for compatibility with clients using old dlib::scoped_ptr
```

Comment:
> With these current changes it looks like people who don't explicitly include the smart pointers header won't get the alias and then their socket code would fail to compile. 

Doesn't 0115f99 fix the problem with sockets? I've missed the sockets at first because CMake didn't want to recompile the socket tests and that came up only on Appveyor. Let me know if there's some other sockets usage and they're still need to be fixed